pkg/gofr/datasource/pubsub/nats/subscription_manager_test.go

```diff
@@ -0,0 +1,210 @@
+}
+
+func TestSubscriptionManager_getOrCreateBuffer(t *testing.T) {
+	sm := newSubscriptionManager(1)
+	topic := "test.topic"
+
+	buffer := sm.getOrCreateBuffer(topic)
+	assert.NotNil(t, buffer)
+	assert.Empty(t, buffer)
+	assert.Equal(t, 1, cap(buffer))
+
+	// Check that the same buffer is returned for the same topic
+	sameBuffer := sm.getOrCreateBuffer(topic)
+	assert.Equal(t, buffer, sameBuffer)
+}
+
+func TestSubscriptionManager_createOrUpdateConsumer(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockJS := NewMockJetStream(ctrl)
+	mockConsumer := NewMockConsumer(ctrl)
+
+	sm := newSubscriptionManager(1)
+	cfg := &Config{
+		Consumer: "test-consumer",
+		Stream: StreamConfig{
+			Stream:     "test-stream",
+			MaxDeliver: 3,
+		},
+	}
+
+	ctx := context.Background()
+	topic := "test.topic"
+
+	mockJS.EXPECT().CreateOrUpdateConsumer(ctx, cfg.Stream.Stream, gomock.Any()).Return(mockConsumer, nil)
+
+	consumer, err := sm.createOrUpdateConsumer(ctx, mockJS, topic, cfg)
+	require.NoError(t, err)
+	assert.Equal(t, mockConsumer, consumer)
+}
+
+func TestSubscriptionManager_consumeMessages(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockConsumer := NewMockConsumer(ctrl)
+	mockLogger := logging.NewMockLogger(logging.DEBUG)
+
+	sm := newSubscriptionManager(1)
+	cfg := &Config{MaxWait: time.Second}
+	topic := "test.topic"
+	buffer := make(chan *pubsub.Message, 1)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mockBatch := createMockMessageBatch(ctrl)
+	mockConsumer.EXPECT().Fetch(gomock.Any(), gomock.Any()).Return(mockBatch, nil).AnyTimes()
+
+	go sm.consumeMessages(ctx, mockConsumer, topic, buffer, cfg, mockLogger)
+
+	select {
+	case msg := <-buffer:
+		assert.NotNil(t, msg)
+		assert.Equal(t, topic, msg.Topic)
+	case <-time.After(2 * time.Second):
+		t.Fatal("Timed out waiting for message")
+	}
+}
+
+func createMockMessageBatch(ctrl *gomock.Controller) jetstream.MessageBatch {
+	mockBatch := NewMockMessageBatch(ctrl)
+	mockMsg := NewMockMsg(ctrl)
+
+	mockMsg.EXPECT().Data().Return([]byte("test message")).AnyTimes()
+	mockMsg.EXPECT().Headers().Return(nil).AnyTimes()
+
+	msgChan := make(chan jetstream.Msg, 1)
+	msgChan <- mockMsg
+	close(msgChan)
+
+	mockBatch.EXPECT().Messages().Return(msgChan).AnyTimes()
+	mockBatch.EXPECT().Error().Return(nil).AnyTimes()
+
+	return mockBatch
+}
+
+func TestSubscriptionManager_Close(t *testing.T) {
+	sm := newSubscriptionManager(1)
+	topic := "test.topic"
+
+	// Create a subscription and buffer
+	ctx, cancel := context.WithCancel(context.Background())
+	sm.subscriptions[topic] = &subscription{cancel: cancel}
+	sm.topicBuffers[topic] = make(chan *pubsub.Message, 1)
+
+	sm.Close()
+
+	assert.Empty(t, sm.subscriptions)
+	assert.Empty(t, sm.topicBuffers)
+
+	// Check that the context was canceled
+	select {
+	case <-ctx.Done():
+		// Expected behavior
+	default:
+		t.Fatal("Context was not canceled")
+	}
```

Comment:
No need for select and channel

If ctx.Err() == nil {
   t.Fatal(
}

Reply:
sure corrected